### PR TITLE
feat(home): integrate youtube data api to fetch video meta data from a playlist

### LINKED ISSUE #320: feat(home): integrate youtube data api to fetch video meta data from a playlist
youtube

[docs](https://developers-dot-devsite-v2-prod.appspot.com/youtube/v3/docs/videos) to list videos by ids 
[docs](https://developers.google.com/youtube/v3/docs/playlistItems) to list items of a playlist

### PR DESCRIPTION
**Pull request by Shurtu-gal**
### Your checklist for this pull request

🚨Please review the [guidelines for contributing](https://github.com/Monday-Morning/project-tahiti/blob/main/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your default branch!
- [x] Make sure you are making a pull request against the **default branch** (left side). Also you should start _your branch_ off _default branch_.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.
- [ ] I have added necessary documentation (if appropriate)

### Linked Issues
Fixes #320 

### Description

Please describe your pull request.

❤️Thank you!

### Post merge checklist

- [x] Follow steps from the [guidelines for contributing](https://github.com/Monday-Morning/project-tahiti/blob/main/CONTRIBUTING.md) to this repository.
- [ ] If you are a new contributor, ping in the thread and one of the maintainers will add you to all-contributors list.

### CODE REVIEW

**Review by 120EE0692 — CHANGES_REQUESTED** [review]

File: client/src/screens/Home.js [hunk 1]
```diff
@@ -52,10 +52,10 @@ function Home({ issues, squiggles }) {
 
         {/* <Banner /> */}
 
-        {/* <div style={{ marginTop: 35 }}> */}
-        {/* <SocialMedia /> */}
-        <Trending />
-        {/* </div> */}
+        <div style={{ marginTop: 35 }}>
+          <SocialMedia youtubeLink={youtubeLink} />
```
Comment (120EE0692) [inline client/src/screens/Home.js:56]: @Shurtu-gal please comments this out.

File: client/src/pages/index.jsx [hunk 2]
```diff
@@ -14,7 +14,7 @@ import Home from '../screens/Home';
 import getLatestIssues from '../graphql/queries/homepage/getLatestIssues';
 import getLatestSquiggle from '../graphql/queries/homepage/getLatestSquiggle';
 
-function HomePage({ issues, squiggles }) {
+function HomePage({ issues, squiggles, YoutubeLink }) {
```
Comment (120EE0692) [inline client/src/pages/index.jsx:17]: @Shurtu-gal use camel case

File: client/src/pages/index.jsx [hunk 3]
```diff
@@ -84,7 +84,11 @@ function HomePage({ issues, squiggles }) {
         <ActivityIndicator size={150} />
       ) : (
         <Marginals>
-          <Home issues={issues} squiggles={squiggles} />
+          <Home
+            issues={issues}
+            squiggles={squiggles}
+            youtubeLink={YoutubeLink}
```
Comment (120EE0692) [inline client/src/pages/index.jsx:90]: use camel case

File: client/src/pages/index.jsx [hunk 4]
```diff
@@ -114,10 +118,20 @@ export async function getStaticProps({ preview }) {
     query: getLatestSquiggle,
   });
 
+  const youtubeResponse = await fetch(
+    `https://youtube.googleapis.com/youtube/v3/playlistItems?part=snippet%2CcontentDetails&maxResults=25&playlistId=${process.env.NEXT_PUBLIC_YOUTUBE_PLAYLIST_ID}&key=${process.env.NEXT_PUBLIC_YOUTUBE_API_KEY}`,
+  );
+  const youtubeData = await youtubeResponse.json();
+  const YoutubeLink = youtubeData.items.map(
```
Comment (120EE0692) [inline client/src/pages/index.jsx:125]: use camel case

File: client/src/pages/index.jsx [hunk 5]
```diff
@@ -114,10 +118,20 @@ export async function getStaticProps({ preview }) {
...
+    (item) =>
+      'https://www.youtube.com/embed/' + item.snippet.resourceId.videoId,
+  );
+
   return {
     props: {
       issues,
       squiggles,
+      YoutubeLink,
```
Comment (120EE0692) [inline client/src/pages/index.jsx:134]: use camel case

**Review by 120EE0692 — CHANGES_REQUESTED** [review]

**Review by 120EE0692 — APPROVED** [review]

**Comment by 120EE0692** [comment]
@rutajdash env keys are used in server side.

File: client/env/.env.sample [hunk 6]
```diff
@@ -34,5 +34,7 @@ REVALIDATE_SECRET_TOKEN=revalidate-secret-token
 # ----- ----- GOOGLE ----- -----
 
 NEXT_PUBLIC_GOOGLE_CLIENT_ID="unique-client-id"
+NEXT_PUBLIC_YOUTUBE_PLAYLIST_ID=some-playlist-id
+NEXT_PUBLIC_YOUTUBE_API_KEY=some-api-key
```
Comment (rutajdash) [inline client/env/.env.sample:38]: Remove `NEXT_PUBLIC_`
As it is used server side, no need to expose credentials to the public.
Refactor in all places in code.

**Review by rutajdash — CHANGES_REQUESTED** [review]
This is good work.

File: client/env/.env.sample (hunk 6 above)
Comment (Shurtu-gal) [reply client/env/.env.sample:38]: Sure

Comment (Shurtu-gal) [reply client/env/.env.sample:38]: > Remove `NEXT_PUBLIC_` As it is used server side, no need to expose credentials to the public. Refactor in all places in code.

Playlist Id should remain public or not?

Comment (120EE0692) [reply client/env/.env.sample:38]: > > Remove `NEXT_PUBLIC_` As it is used server side, no need to expose credentials to the public. Refactor in all places in code.
> 
> Playlist Id should remain public or not?

Playlist id is not public.

File: client/src/components/homepage/SocialMedia.js [hunk 7]
```diff
@@ -16,17 +15,22 @@ import insta4 from '../../assets/images/instagram/insta4.jpeg';
...
+const Header = ({ Icon, title, buttonText, onClick, ...color }) => {
+  const classes = useStyles(color);
+=======
 const Header = ({ Icon, title, buttonText, onClick }) => {
   const classes = useStyles();
+>>>>>>> 8c65f8b (feat: add youtube api)
+=======
+const Header = ({ Icon, title, buttonText, onClick, ...color }) => {
+  const classes = useStyles(color);
+>>>>>>> eb75993 (fix: styling issues, camelCase)
```
Comment (rutajdash) [inline client/src/components/homepage/SocialMedia.js:33]: Same

File: client/src/components/homepage/SocialMedia.js [hunk 8]
```diff
@@ -16,17 +15,22 @@ import insta4 from '../../assets/images/instagram/insta4.jpeg';
...
-  'https://www.youtube.com/embed/_KJlm1mafGo',
-  'https://www.youtube.com/embed/btc-_kqkdpc',
-  'https://www.youtube.com/embed/c6SglPpwmVg',
-];
-
+<<<<<<< HEAD
+<<<<<<< HEAD
+const Header = ({ Icon, title, buttonText, onClick, ...color }) => {
+  const classes = useStyles(color);
+=======
```
Comment (rutajdash) [inline client/src/components/homepage/SocialMedia.js:26]: Why were unresolved conflicts pushed?

File: client/src/components/homepage/SocialMedia.js [hunk 9]
```diff
@@ -51,7 +55,15 @@ const Header = ({ Icon, title, buttonText, onClick }) => {
...
-const Socials = () => {
+<<<<<<< HEAD
+<<<<<<< HEAD
+const Socials = ({ youtubeLink }) => {
+=======
+const Socials = ({ YOUTUBE_LINKS }) => {
+>>>>>>> 8c65f8b (feat: add youtube api)
+=======
+const Socials = ({ youtubeLink }) => {
+>>>>>>> eb75993 (fix: styling issues, camelCase)
```
Comment (rutajdash) [inline client/src/components/homepage/SocialMedia.js:66]: Same

File: client/src/pages/index.jsx [hunk 10]
```diff
@@ -138,10 +139,20 @@ export async function getStaticProps({ preview }) {
     variables: { categoryNumbers: 62, limit: 1 },
   });
 
+  const youtubeResponse = await fetch(
+    `https://youtube.googleapis.com/youtube/v3/playlistItems?part=snippet%2CcontentDetails&maxResults=25&playlistId=${process.env.YOUTUBE_PLAYLIST_ID}&key=${process.env.YOUTUBE_API_KEY}`,
+  );
+  const youtubeData = await youtubeResponse.json();
+  const youtubeLink = youtubeData.items.map(
```
Comment (rutajdash) [inline client/src/pages/index.jsx:146]: If this variable contains an array of data, make it a plural like `youtubeLinks` and then refactor everywhere.

**Review by rutajdash — CHANGES_REQUESTED** [review]

**Review by rutajdash — APPROVED** [review]

**Comment by Shurtu-gal** [comment]
> @Shurtu-gal please resolve conflicts

Sure

**Review by rutajdash — APPROVED** [review]

**Pull request merged**